rules/aip0158/response_plural_first_field.go

```diff
@@ -30,7 +31,16 @@ var responsePluralFirstField = &lint.MessageRule{
 		// Throw a linter warning if, the first field in the message is not named
 		// according to plural(message_name.to_snake().split('_')[1:-1]).
 		firstField := m.GetFields()[0]
-		want := utils.ToPlural(firstField.GetName())
+
+		// If the field is a resource, use the `plural` annotation to decide the
+		// appropriate plural field name.
+		var want string
+		if res := utils.GetResource(firstField.GetMessageType()); res != nil && res.GetPlural() != "" {
+			want = strcase.SnakeCase(res.GetPlural())
+		} else {
+			want = utils.ToPlural(firstField.GetName())
```

Comment:
this already existed of course, but I feel like pluralizing the resource singular or the type would be more robust than the field name.

Reply:
I think this is meant to support the scenario where the paginated field isn't a resource. I agree with you, but I don't know if we should change it right now. There are maybe one or two Standard List-like fields that paginate a `repeated string` field that I know of. This rule being in AIP-158, not AIP-132 is poignant.

Reply:
Opened #1160